govcd/nsxt_firewall_group.go

```diff
@@ -219,7 +231,7 @@ func (firewallGroup *NsxtFirewallGroup) Update(firewallGroupConfig *types.NsxtFi
 // Delete allows users to delete NSX-T Firewall Group
 func (firewallGroup *NsxtFirewallGroup) Delete() error {
 	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointFirewallGroups
-	minimumApiVersion, err := firewallGroup.client.checkOpenApiEndpointCompatibility(endpoint)
+	apiVersion, err := firewallGroup.client.getOpenApiHighestElevatedVersion(endpoint)
```

Comment:
Just being cautious: after changing the way we chose API versions across this PR, does it work with the latest 10.4?

Reply:
Testing will cover 10.4, but if you remember - we introduced "smart" API elevation back with PR #399. 

The point of it is that it doesn't "dumbly" just pick the latest API version available. We only pick "highest supported version" from our predefined list and this PR does it here: https://github.com/vmware/go-vcloud-director/pull/487/files#diff-d48dd6aa67017584bb6e37e8e459a46748e9f2660ff686da9a84b3bb18449792R99-R102

```go
types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointFirewallGroups: {
   //"34.0", // Basic minimum required version
   "36.0", // Adds support for Dynamic Security Groups by deprecating `Type` field in favor of `TypeValue`
},
```


This definition ensures that we will not elevate API above v36.0 even if v39.0 is supported in the future. So unless VCD itself breaks API V36.0 - we are good to go. 

We only elevate to known and tested API versions.